### PR TITLE
feat(auth): add reducers

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: libs/auth/src/reducers/auth-initial-state.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { DaffAuthReducerState } from './auth-reducer-state.interface';
+import { DaffAuthToken } from '../models/auth-token';
+
+export const daffAuthInitialState: DaffAuthReducerState<any> = Object.freeze({
```
Comment (damienwebdev) [inline libs/auth/src/reducers/auth-initial-state.ts:4]: Why freeze the Object this is inconsistent with other packages?

File: libs/auth/src/reducers/auth-reducer-state.interface.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { DaffAuthToken } from '../models/auth-token';
+
+export interface DaffAuthReducerState<T extends DaffAuthToken> {
```
Comment (damienwebdev) [inline libs/auth/src/reducers/auth-reducer-state.interface.ts:3]: I think this model is wrong. "login" and "register" probably have split "Loadable" states (errors + loading).

Comment (griest024) [reply libs/auth/src/reducers/auth-reducer-state.interface.ts:3]: that makes sense

File: libs/auth/src/reducers/auth-initial-state.ts (hunk 1 above)
Comment (griest024) [reply libs/auth/src/reducers/auth-initial-state.ts:4]: Unless we specifically want to allow the initial state to be modified it seems safest to freeze it.

Comment (griest024) [reply libs/auth/src/reducers/auth-initial-state.ts]: Removed `Object.freeze` from the initial states

**Pull request merged**